### ISSUE TITLE
Problem with setting up 

### ISSUE THREAD

**Issue by FireworkZhao** [CNS-OIST/STEPS#38]
After pulling up the docker file, user can't successfully proceed with "make"
```
/usr/include/c++/12/bits/hashtable_policy.h:1277:25: error: static assertion failed: hash function must be invocable with an argument of key type
 1277 |           static_assert(__is_invocable<const _Hash&, const _Kt&>{},
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/hashtable_policy.h:1277:25: note: ‘std::__is_invocable<const steps::util::impl::hash_ref<std::hash<double> >&, const count_copies<float>&>()’ evaluates to false
/usr/include/c++/12/bits/hashtable_policy.h:1279:27: error: no match for call to ‘(const steps::util::impl::hash_ref<std::hash<double> >) (const count_copies<float>&)’
 1279 |           return _M_hash()(__k);
      |                  ~~~~~~~~~^~~~~
/STEPS/src/steps/util/collections.hpp:53:38: note: candidate: ‘template<class V> size_t steps::util::impl::hash_ref<H>::operator()(std::reference_wrapper<_Functor>) const [with H = std::hash<double>]’
   53 |         template <typename V> size_t operator()(std::reference_wrapper<V> v) const { return hasher(v.get()); }
      |                                      ^~~~~~~~
/STEPS/src/steps/util/collections.hpp:53:38: note:   template argument deduction/substitution failed:
/usr/include/c++/12/bits/hashtable_policy.h:1279:27: note:   ‘count_copies<float>’ is not derived from ‘std::reference_wrapper<_Tp>’
 1279 |           return _M_hash()(__k);
      |                  ~~~~~~~~~^~~~~
make[2]: *** [test/unit/cpp/CMakeFiles/test_membership.dir/build.make:76: test/unit/cpp/CMakeFiles/test_membership.dir/test_membership.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:512: test/unit/cpp/CMakeFiles/test_membership.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

**Comment by tristan0x** [comment]
Thank you for your feedback @FireworkZhao and sorry for the late reply.
I am confused because either you clone the GitHub STEPS repository to compile the project (with CMake and `make` for instance), or you use Docker to pull an image that contains an installation of STEPS (already compiled).
Here you mention a Docker and `make` at the same time.

Can you please mention the list of commands (with their console output) you executed to reach this error to understand the situation?

**Comment by FireworkZhao** [comment]
Thank you so much for your patience. I have done both seperatly.
In the case of pullling Docker image, I wonder is it possilbe to start a  new simulation with my own meshes on Juypter Notebook? or I can only testing the examples on the user manual?
In the case of compling the project pulling from Github, I found it nearly impossible to do it on Windows hence I moved it to a server with Linux system and solved the setting-up problem. The result showed as below, which I believed suggesting successful installation:
(venv)  ....@....... :~/STEPS$ python3 -c "import steps; steps._greet()"

STochastic Engine for Pathway Simulation
Version:  3.5.0
License:  GPL2.0
Website:  steps.sourceforge.net
CXX Binding: Cython

This leads to the next problem I proposed afterwards.
